commit 61f4fb541504341ae674a1744a8baacc51bd1109
Author: Twaik Yont <9674930+twaik@users.noreply.github.com>
Date:   Wed Feb 18 21:33:21 2026 +0200

    Fix R8 shrink breakage (JNI + reflection)
    
    Add keep rules for JNI-registered natives, MainActivity methods, CmdEntryPoint.main, and Preference.onSetInitialValue.
    Switch to getDeclaredMethod + setAccessible(true) for protected Preference method.
    
    Signed-off-by: Twaik Yont <9674930+twaik@users.noreply.github.com>
---
 app/proguard-rules.pro                             | 38 +++++++++++-----------
 .../main/java/com/termux/x11/LoriePreferences.java |  4 +--
 2 files changed, 21 insertions(+), 21 deletions(-)

diff --git a/app/proguard-rules.pro b/app/proguard-rules.pro
index 55835b3..25d5348 100644
--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -1,23 +1,23 @@
-# Add project specific ProGuard rules here.
-# You can control the set of applied configuration files using the
-# proguardFiles setting in build.gradle.
-#
-# For more details, see
-#   http://developer.android.com/guide/developing/tools/proguard.html
+-keepattributes SourceFile,LineNumberTable
 
-# If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
-# class:
-#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
-#   public *;
-#}
+-dontobfuscate
 
-# Uncomment this to preserve the line number information for
-# debugging stack traces.
-#-keepattributes SourceFile,LineNumberTable
+# Keep classes/members referenced from JNI (FindClass/GetMethodID/RegisterNatives) + main entrypoint
+-keep class com.termux.x11.LorieView {
+    native <methods>;
+    void resetIme();
+}
 
-# If you keep the line number information, uncomment this to
-# hide the original source file name.
-#-renamesourcefileattribute SourceFile
+-keep class com.termux.x11.MainActivity {
+    public static com.termux.x11.MainActivity getInstance();
+    void clientConnectedStateChanged();
+}
 
--dontobfuscate
+-keep class com.termux.x11.CmdEntryPoint {
+    public static void main(java.lang.String[]);
+}
+
+# Keep Preference subclasses' onSetInitialValue for reflective calls
+-keepclassmembers class * extends android.preference.Preference {
+    void onSetInitialValue(boolean, java.lang.Object);
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/termux/x11/LoriePreferences.java b/app/src/main/java/com/termux/x11/LoriePreferences.java
index 332575c..d1a0739 100644
--- a/app/src/main/java/com/termux/x11/LoriePreferences.java
+++ b/app/src/main/java/com/termux/x11/LoriePreferences.java
@@ -191,8 +191,8 @@ public class LoriePreferences extends AppCompatActivity implements PreferenceFra
         private static final Method onSetInitialValue;
         static {
             try {
-                //noinspection JavaReflectionMemberAccess
-                onSetInitialValue = Preference.class.getMethod("onSetInitialValue", boolean.class, Object.class);
+                onSetInitialValue = Preference.class.getDeclaredMethod("onSetInitialValue", boolean.class, Object.class);
+                onSetInitialValue.setAccessible(true);
             } catch (NoSuchMethodException e) {
                 throw new RuntimeException(e);
             }
